src/test/java/com/google/sps/comments/CommentServletTest.java

```diff
@@ -0,0 +1,167 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.google.sps.servlets;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import static org.mockito.Mockito.when;
+import org.mockito.Mockito;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+
+import com.google.sps.testing.GoodDeed;
+import com.google.sps.testing.GoodDeedsServlet;
+import com.google.sps.data.Comment;
+import com.google.sps.servlets.CommentServlet;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import com.google.gson.Gson;
+
+import java.util.List;
+import java.util.ArrayList;
+
+@RunWith(JUnit4.class)
+public class CommentServletTest {
+    //Properties and Entities 
+    private static final String COMMENTS = "Comments";
+    private static final String COMMENT_FORM_NAME = "newComment";
+    private static final String DEED_ENTITY = "GoodDeed";
+    private static final String DAILY_DEED = "Daily Deed";
+    private static final String DESCRIPTION = "Description";
+    private static final String LINK = "Link";
+    private static final String NAME = "Name";
+    
+    //Inputs for the properties and entities
+    private static final List COMMENTS_LIST = new ArrayList<>();
```

Comment:
`COMMENTS_LIST` shouldn't be defined here based on the usage. This will lead to issues when you use this list in more than one test. `setUp()` will be called before every test and will add `NEW_COMMENT` to the list each time. Just move the declaration to `testGet()`

Reply:
Sounds good!